src/applications/personalization/profile360/containers/Vet360ProfileField.jsx

```diff
@@ -0,0 +1,206 @@
+    }
+
+    return (
+      <div className="vet360-profile-field">
+        <HeadingWithEdit onEditClick={this.isEditLinKVisible() && onEdit}>{title}</HeadingWithEdit>
+        {isEditing && this.renderEditModal()}
+        {content}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  const { fieldName } = ownProps;
+  const {
+    user: {
+      profile: {
+        vet360: {
+          [fieldName]: existingFieldData
+        }
+      }
+    },
+    vaProfile: {
+      modal: currentlyOpenEditModal,
+      formFields: {
+        [fieldName]: existingFieldValue
+      }
+    },
+
+    vet360: {
+      transactions,
+      fieldTransactionMap
+    }
+  } = state;
+
+  const fieldTransactionData = fieldTransactionMap[fieldName];
+  let transaction = null;
+
+  if (fieldTransactionData && fieldTransactionData.transactionId) {
+    transaction = transactions.find(t => t.data.attributes.transactionId === fieldTransactionData.transactionId);
+  }
+
+  return {
+    data: existingFieldData,
+    field: existingFieldValue,
+    isEditing: currentlyOpenEditModal === fieldName,
+    transaction,
+    transactionRequest: fieldTransactionData
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  // TODO turn analytics back on later
+  /* eslint-disable no-unused-vars */
+
+  const {
+    fieldName,
+    analyticsSectionName: sectionName
+  } = ownProps;
+
+  const captureEvent = (actionName) => {
+    // if (sectionName && actionName) {
+    //   recordEvent({
+    //     event: 'profile-navigation',
+    //     'profile-action': actionName,
+    //     'profile-section': sectionName,
+    //   });
+    // }
+  };
+
+  const closeEditModal = () => dispatch(openModal(null));
+  const openEditModal = () => dispatch(openModal(fieldName));
+
+  return {
+    clearErrors() {
+      dispatch(clearErrors);
+    },
+
+    getTransactionStatus(transaction) {
+      dispatch(getTransactionStatus(transaction, fieldName));
+    },
+
+    onAdd() {
+      captureEvent('add-link');
+      openEditModal();
+    },
+
+    onCancel() {
+      captureEvent('cancel-button');
+      closeEditModal();
+    },
+
+    onChange(...args) {
+      dispatch(updateFormField[fieldName](...args));
+    },
+
+    onEdit() {
+      captureEvent('edit-link');
+      openEditModal();
+    },
+
+    onSubmit(...args) {
+      captureEvent('update-button');
+      dispatch(saveField[fieldName](...args));
+    }
+
+  };
+};
+
+const Vet360ProfileFieldContainer = connect(mapStateToProps, mapDispatchToProps)(Vet360ProfileField);
+
+Vet360ProfileFieldContainer.propTypes = {
+  analyticsSectionName: PropTypes.string.isRequired,
+  fieldName: PropTypes.oneOf(Object.values(VET360.FIELD_NAMES)).isRequired,
+  renderContent: PropTypes.func.isRequired,
+  renderEditModal: PropTypes.func.isRequired,
+  title: PropTypes.string.isRequired
+};
+
+export default Vet360ProfileFieldContainer;
+export { Vet360ProfileField };
```

Comment:
This is the main point of all of the changes - consolidating all of the common 360 field behavior into a container.